### PR TITLE
Fix broken URL fragments

### PR DESCRIPTION
**Pull request by OnkarRuikar**

### CODE REVIEW

File: files/en-us/learn/forms/basic_native_form_controls/index.md [hunk 1]
```diff
@@ -125,7 +125,7 @@ A checkbox is created using the {{HTMLElement("input")}} element with a [`type`]
 <input type="checkbox" id="questionOne" name="subscribe" value="yes" checked />
 ```
 
-Related checkbox items should use the same [`name`](/en-US/docs/Web/HTML/Element/input#name) attribute. Including the [`checked`](/en-US/docs/Web/HTML/Element/input/checkbox#checked) attribute makes the checkbox checked automatically when the page loads. Clicking the checkbox or its associated label toggles the checkbox on and off.
+Related checkbox items should use the same [`name`](/en-US/docs/Web/HTML/Element/input#name) attribute. Including the [`checked`](/en-US/docs/Web/HTML/Element/input/checkbox#attr-checked) attribute makes the checkbox checked automatically when the page loads. Clicking the checkbox or its associated label toggles the checkbox on and off.
```
Comment (teoli2003) [inline files/en-us/learn/forms/basic_native_form_controls/index.md:128]: The old one (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#checked ) is working, so I would keep it.

```suggestion
Related checkbox items should use the same [`name`](/en-US/docs/Web/HTML/Element/input#name) attribute. Including the [`checked`](/en-US/docs/Web/HTML/Element/input/checkbox#checked) attribute makes the checkbox checked automatically when the page loads. Clicking the checkbox or its associated label toggles the checkbox on and off.
```

File: files/en-us/learn/forms/other_form_controls/index.md [hunk 2]
```diff
@@ -51,7 +51,7 @@ This renders like so:
 
 The main difference between a `<textarea>` and a regular single-line text field is that users are allowed to include hard line breaks (i.e. pressing return) that will be included when the data is submitted.
 
-`<textarea>` also takes a closing tag, and any default text you want it to contain should be put between the opening and closing tags. In contrast, the {{HTMLElement("input")}} is a {{glossary("void element")}} with no closing tag — any default value is put inside the [`value`](/en-US/docs/Web/HTML/Element/input#value) attribute.
+`<textarea>` also takes a closing tag, and any default text you want it to contain should be put between the opening and closing tags. In contrast, the {{HTMLElement("input")}} is a {{glossary("void element")}} with no closing tag — any default value is put inside the [`value`](/en-US/docs/Web/HTML/Element/input) attribute.
```
Comment (teoli2003) [inline files/en-us/learn/forms/other_form_controls/index.md:54]: This one is working: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#value

```suggestion
`<textarea>` also takes a closing tag; any default text you want it to contain should be put between the opening and closing tags. In contrast, the {{HTMLElement("input")}} is a {{glossary("void element")}} with no closing tag — any default value is put inside the [`value`](/en-US/docs/Web/HTML/Element/input#value) attribute.
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
Two over-corrections, but else this is good to go.

**Comment by OnkarRuikar** [comment]
Actually both IDs are correct:
bug

The script picked it because `htmlattrdef` macro generates `attr-*` IDs.

Keeping both the IDs(`#checked` vs `#attr-checked`) for same location will be hard to maintain in long term. I think using only def list ids will be more stable approach.
I don't know why the macro generates new ID `attr-*` when the def list already generates them. Why only [HTML pages have self linking def lists](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#additional_attributes)  and other page types like [CSS don't have such feature](https://developer.mozilla.org/en-US/docs/Web/CSS/background-size#values)?

/cc @caugner 
***
**Note:** I am working on a maintenance script for URL fragments. Because every time a PR updates a markdown header (`### Header`) there is a high chance that bunch of fragments break all over the repo. At the moment there are 900+ broken fragments in the repo.

**Review by teoli2003 — APPROVED** [review]
Thank you!

**Comment by teoli2003** [comment]
> Keeping both the IDs(`#checked` vs `#attr-checked`) for same location will be hard to maintain in long term. I think using only def list ids will be more stable approach. I don't know why the macro generates new ID `attr-*` when the def list already generates them.

I agree; the reason why we have the macro adding `attr-*` IDs is that both in the Wiki era and in the early Yari era, definition lists didn't get IDs automatically. This has been changed and the two macros are no more useful. We should get rid of them.

**Pull request merged**